src/LEGO.AsyncAPI.Readers/ParseNodes/MapNode.cs

```diff
@@ -196,7 +196,7 @@ public string GetReferencePointer()
                 return null;
             }
 
-            return refNode.GetScalarValue();
+            return refNode.AsValue().GetScalarValue();
```

Comment:
This `AsValue` already has a properly worded exception if it's the wrong type. Before inside of `GetScalerValue` it would throw a rather vague exception. I changed GetScalerValue to only work with `NodeValue`.